modin/pandas/dataframe.py

```diff
@@ -364,41 +364,15 @@ def apply(
         )
         if not isinstance(query_compiler, type(self._query_compiler)):
             return query_compiler
-        # This is the simplest way to determine the return type, but there are checks
-        # in pandas that verify that some results are created. This is a challenge for
-        # empty DataFrames, but fortunately they only happen when the `func` type is
-        # a list or a dictionary, which means that the return type won't change from
-        # type(self), so we catch that error and use `type(self).__name__` for the return
-        # type.
-        try:
-            if axis == 0:
-                init_kwargs = {"index": self.index}
-            else:
-                init_kwargs = {"columns": self.columns}
-            return_type = type(
-                getattr(pandas, type(self).__name__)(**init_kwargs).apply(
-                    func,
-                    axis=axis,
-                    raw=raw,
-                    result_type=result_type,
-                    args=args,
-                    **kwargs,
-                )
-            ).__name__
-        except Exception:
-            return_type = type(self).__name__
-        if return_type not in ["DataFrame", "Series"]:
-            return query_compiler.to_pandas().squeeze()
+
+        reduced_index = pandas.Index(["__reduced__"])
+        if query_compiler.get_axis(axis).equals(
+            reduced_index
+        ) or query_compiler.get_axis(axis ^ 1).equals(reduced_index):
+            result = Series(query_compiler=query_compiler)
         else:
-            result = getattr(sys.modules[self.__module__], return_type)(
-                query_compiler=query_compiler
-            )
-            if isinstance(result, Series):
-                if axis == 0 and result.name == self.index[0] or result.name == 0:
-                    result.name = None
-                elif axis == 1 and result.name == self.columns[0] or result.name == 0:
-                    result.name = None
```

Comment:
This logic were ensuring that we always return an unnamed Series. Since this PR, all of the `apply()` results has an axis named `__reduced__` if the QC stores a Series, this will [automatically ensure](https://github.com/modin-project/modin/blob/ab2855b58cadcc769575103a9844544a69628e84/modin/pandas/series.py#L165-L176) that the Series will be unnamed.